core-java/src/test/java/com/baeldung/deepcopy/DeepCopyTest.java

```diff
@@ -0,0 +1,131 @@
+package com.baeldung.deepcopy;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertNotSame;
+
+import java.io.IOException;
+
+import org.apache.commons.lang.SerializationUtils;
+import org.junit.Test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+
+public class DeepCopyTest {
+
+    @Test
+    public void whenCreatingDeepCopyWithCopyConstructor_thenObjectsShouldNotBeSame() {
+
+        Address address = new Address("Downing St 10", "London", "England");
+        User pm = new User("Prime", "Minister", address);
+        User deepCopy = new User(pm);
+
+        assertNotSame(deepCopy, pm);
```

Comment:
Use the AssertJ assertion:   assertThat(deepCopy).isNotSameAs(pm)

Reply:
Applied